[stable/traefik] Add support for proxyProtocol.

timstoop opened the pull request:
This adds support for proxyProtocol, which allows traefik to present actual user IPs when behind a loadbalancer like EBS.

timothyclarke committed:
Made MariaDB Optional, Also added optional uri path

timothyclarke committed:
Adding mariadb secrets which was provided by the subchart when subchart is not run

Commit:
Add support for proxyProtocol.

Commit:
[incubator/kibana] add Kibana chart (#2719)

[stable/kibana] add Kibana chart

timothyclarke committed:
Bumped Wordpress chart number after rebase

timothyclarke committed:
Updated wordpress/requirements.lock to match yaml

Commit:
Update README

Remove extra word

Commit:
Update version on Chart.yaml

jstriebel committed:
[stable/kibana] fix chart name in README (#2947)

* [stable/kibana] fix chart name in README

* bump chart version

timothyclarke committed:
Externalised db details per request of @prydonius

timothyclarke committed:
Testing db connection only when deploying embedded db

kachkaev committed:
[stable/traefik] Bump Traefik version to 1.4.5 (#2948)

andor-pierdelacabeza committed:
Update README to match values.yaml (#2628)

* Update README to match values.yaml

Readme doesn't reflect correctly default values contained on values.yaml

* Upgrade chart version to fix lint

arthur-c committed:
[stable/rabbitmq] Rename RABBITMQ_VHOST (#2853)

* Rename RABBITMQ_VHOST

Rename RABBITMQ_VHOST to RABBITMQ_DEFAULT_VHOST to work with latest rabbitmq:3 images. 
Source: https://hub.docker.com/_/rabbitmq/

* Bump rabbitmq chart version

tomgco committed:
[stable/external-dns] Fixing incorrect parameter name (#2841)

unguiculus commented:
cc @krancour

egeland committed:
[stable/nginx-ingress] allow for additional environment variables to be set (#2808)

egeland committed:
[stable/nginx-ingress] allow custom revisionHistoryLimit to be set (#2811)

terjesannum committed:
[stable/kube-state-metrics] Add support for pod annotations (#2849)

ebabani committed:
[stable/nginx-ingress] Add lifecycle hooks to the deployment template (#2855)

krancour commented on stable/traefik/templates/configmap.yaml:
```diff
@@ -30,6 +39,15 @@ data:
+              {{- $element | quote }}
+          {{- end -}}
+          ]
+      {{- end }}
```
Can this bit go into a helper or something since it's duplicated in two places?

timstoop replied:
I'd love to, but I'm afraid I'm going to need some guidance with that.

krancour replied:
Have a look at `_helpers.tpl` for examples of some helpers that are already used in this chart.

timstoop replied:
I did, there's only one in there and it does a string replace. I'm looking for an example of how to replace an entire block of code. I'll keep looking in the other charts (tomorrow, it's 1.30am here).

krancour commented on stable/traefik/Chart.yaml:
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: traefik
-version: 1.14.2
+version: 1.14.3
```
Since this adds a feature, it should be the minor version number that is incremented.

krancour commented:
Love the feature. Mad one comment about extracting some duplicated logic into a helper.

This will also need some updates to the README. There's a table in there that goes over all config options in detail. And It wouldn't actually hurt to have a who separate section of the documentation dedicated to this-- or at least point to the relevant Traefik docs. PROXY protocol is something that needs to be enabled on _both_ the proxy and the target to work. If applied in one place and not the other things will break. Just need to be sure people are aware.

And again-- I _love_ this addition.

Commit:
Modify version, increase minor due to feature add.

Commit:
Add a description regarding ProxyProtocol.

Commit:
Add a description regarding ProxyProtocol.

Commit:
Explain the new options.

Commit:
Explain the new options.

krancour commented on stable/traefik/README.md:
```diff
@@ -98,6 +98,8 @@ The following tables lists the configurable parameters of the Traefik chart and
 | `nodeSelector`                  | Node labels for pod assignment                                       | `{}`                                      |
 | `tolerations`                   | List of node taints to tolerate                                      | `[]`                                      |
+| `proxyProtocol.enabled`         | Enable receiving the Proxy protocol instead http or https            | `false`                                   |
+| `proxyProtocol.trustedIPs`      | List of trusted ip (ranges) from which to allow Proxy protocol traffic | `["10.0.0.0/8"]`                        |
```
I would reword the descriptions above for better clarity...

Propose changing

> Enable receiving the Proxy protocol instead http or https

to just:

> Enable PROXY protocol support.

Maybe add a link to the section where you elaborate on it?

Propose changing

> List of trusted ip (ranges) from which to allow Proxy protocol traffic

to

> List of proxy IPs (CIDR ranges) trusted to accurately convey the end-user IP.

krancour replied:
The default of `10.0.0.0/8` isn't going to work for lots and lots of people. It might be better to _not_ have a default and make chart install fail if it's not defined. This can be done with the some conditional logic and the [fail function from Sprig](http://masterminds.github.io/sprig/flow_control.html) (which Helm supports).

krancour commented on stable/traefik/README.md:
```diff
@@ -179,3 +181,7 @@ dashboard:
+
+### Proxy Protocol
+
+Normally, when traefik lives behind a LoadBalancer (like an AWS ELB), incoming traffic is all originating from that LoadBalancer, from traefik's point of view.  By setting the ELB to `Proxy` ([via an annotation](https://kubernetes.io/docs/concepts/services-networking/service/#proxy-protocol-support-on-aws)) and telling traefik to expect the Proxy protocol, the ELB acts as a transparant proxy, ie. ...
```
s/traefik/Traefik/

krancour replied:
This section might need more rewording for clarity.

Commit:
Documentation changes, as suggested.

Commit:
Documentation changes, as suggested.

Commit:
change etcd-operator version from v0.3.3 to v0.6.1 (#2863)

rendhalver committed:
Promote docker-registry to stable (#2944)

rjkernick committed:
[stable/sonatype-nexus] Add resource limits/requests to deployment config (#2732)

rendhalver committed:
[stable/kibana] Upgrade Kibana to 6.0.0 (#2953)

tompizmor committed:
[stable/drupal] allow use of external database (#2817)

* Allow Drupal helm chart to use an existing database

* Do not deploy mariadb chart if using external database

* Add new properties to README

* Refactor properties to configure an external database

* Update dependencies

Commit:
[stable/drupal] Release 0.11.1 (#2975)

* drupal: update to `bitnami/drupal:8.4.3-r0`

* drupal: bump chart appVersion to `8.4.3`

* drupal: bump chart version to `0.11.1`

Commit:
[stable/phabricator] Release 0.5.3 (#2968)

* phabricator: update to `bitnami/phabricator:2017.48.0-r0`

* phabricator: bump chart appVersion to `2017.48.0`

* phabricator: bump chart version to `0.5.3`

Commit:
[stable/ghost] Release 2.1.2 (#2967)

* ghost: update to `bitnami/ghost:1.18.4-r0`

* ghost: bump chart appVersion to `1.18.4`

* ghost: bump chart version to `2.1.2`

Commit:
[stable/mongodb] Release 0.4.19 (#2945)

* mongodb: update to `bitnami/mongodb:3.6.0-r0`

* mongodb: bump chart appVersion to `3.6.0`

* mongodb: bump chart version to `0.4.19`

jlaswell committed:
Update chaoskube readme to latest version (#2963)

* Update chaoskube readme to latest version

* Increment Chart.yaml version

timstoop committed:
[stable/traefik] Improve kubernetes section in values.yaml. (#2823)

* Improve kubernetes section in values.yaml.

The values.yaml had the kubernetes section under the ingress section,
which is not correct.

* Fix spaces.

* Increase Chart version.

* Comment out the kubernetes section entirely.

* Bumping chart version

caarlos0 committed:
Fixes on the traefik - prometheus config (#2962)

* buckets should be an array

* all array items should be of the same type

* bump chart version

jainishshah17 committed:
Updated artifactory to version 5.6.2 (#2868)

Updated artifactory to version 5.6.2

alxrem committed:
[stable/grafana] Added value server.service.externalIPs (#2880)

[stable/grafana] Added value server.service.externalIPs

notxcain committed:
[stable/metabase] Make plugins directory configurable (#2722)

* Make plugins directory configurable

* Make linter happy

* Update README.md

anapsix committed:
support adding optional ingress path (#2688)

some ingress' (e.g. Traefik) can benefit from "paths.path" set,
  in addition to "paths.backend" - this change allows adding path

alebabai committed:
[stable/postgresql] Add image pull secrets to deployment template (#2831)

* [stable/postgresql] Add image pull secrets to deployment template

* update deployment template
* add new parameter to README

* [stable/postgresql] bump chart version

update version to 0.8.5

* [stable/postgresql] add an example of imagePullSecrets to values.yaml

alebabai committed:
[incubator/cassandra] Add image pull secrets to statefulset template (#2832)

* [incubator/cassandra] Add image pull secrets to statefulset template

* update statefulset template
* add new parameter to README

* [incubator/cassandra] bump chart version

update version to 0.1.7

* [incubator/cassandra] add an example of ImagePullSecrets to values.yaml

cgebe committed:
[stable/mongodb] Random passwords for mongodb if they are not set in values.yaml (#2850)

* random passwords for mongodb if not set

* increase Chart version to 0.4.19

* README.md adjusted

* version 0.4.19 to 0.4.20

floriantraber committed:
Add nodeSelector (#2938)

smelchior committed:
add eventer to heapster chart (#1998)

erickt committed:
[stable/cluster-autoscaler] Bump to 1.0.3, allow patching pod status (#2776)

This updates the autoscaler to 1.0.3, and adds a few new permissions
that were implemented by Kops:

* https://github.com/kubernetes/kops/pull/3145 - Fixing clusterautoscaler
  rbac.
* https://github.com/kubernetes/kops/pull/3419 - Allow cluster autoscaler
  to scale down.

khrisrichardson committed:
[stable/external-dns] enable freeform env via secrets, doc cleanup (#2356)

* enable freeform env via secrets, doc cleanup

* Bump version

Fsero committed:
[stable/nginx-ingress] add HPA in case of Deployment (#2736)

DavidHuie committed:
[stable/consul] Ingress for Consul UI (#2408)

This introduces an optional ingress record for Consul UI.

kujenga committed:
[incubator/vault] various improvements aimed at production use (#2848)

* incubator/vault: add annotations support to service

This allows various customizations of the behavior of the service,
including the use of GKE internal load balancing [1]

[1] https://cloud.google.com/kubernetes-engine/docs/internal-load-balancing

* incubator/vault: allow injection of custom secrets, facilitating tls and more

This commit allows users of the chart to inject customized secrets
that are needed to leverage various pieces of Vault functionality,
including TLS and storage backends like GCS that require a file.

* incubator/vault: upgrade to latest version 0.9.0

* incubator/vault: bump version for annotations, secrets mounting, upgraded container

* incubator/vault: Update README with new version, customSecrets capability

* incubator/vault: fix comment spacing based on lint failures

timurkhafizov committed:
[stable/external-dns] Update README (#2961)

* [stable/external-dns] Update README to list a proper arguments

* [stable/external-dns] Bump chart patch version number

joosthofman committed:
[stable/Sonarqube] Config map and plugin installation (#2781)

* Created a configmap for sonar properties. Created a install plugin init container to install plugins specified in the values. Fixed static port number in test-config

* Set version to 0.3.0 in sonarqube chart

* Set default persistence from postgres to false

* Fix wrong init container docker image

* fix missing spaces in comments

* Removed persistence false on Postgres DB

* Fix new line error in config, deployment has existingClaim support

* add basic labels to config and install-plugins. fixed storage class value.

ledor473 committed:
[incubator/jaeger] Ingress issue (#2845)

* service.externalPort is not defined

* serviceName is not in line with the Query service name

tamalsaha committed:
Update app version for Swift (#2872)

hairmare committed:
[stable/sonatype-nexus] Fix default in sonatype nexus readme (#2924)

* [stable/sonatype-nexus] Update config in README.md

Update the configuration in README.md to reflect the defaults set
in values.yaml.

* [stable/sonatype-nexus] Update version in Chart.yaml

stackedsax committed:
Provide a parameter to enable Thrift (#2932)

timothyclarke committed:
stable/wordpress: Updated externaldb variable names per request

pulledtim committed:
Added new parameter fullnameOverride

hairmare committed:
[stable/sonarqube]: Do not set targetPort for ClusterIP service (#2874)

[stable/sonarqube]: Do not set targetPort for ClusterIP service

Limess committed:
Neo4j: add sidecar container option for core pod (#2829)

[stable/neo4j] add optional sidecar container configuration for core pod.

acartapanis committed:
Supports for existing volume claim (#2984)

Supports for existing volume claim

Commit:
Remove Spinnaker from e2e tests (#2990)

irabinovitch committed:
[datadog] Update Mainainers (#2980)

* [datadog] Update Mainainers

Updating maintainer list per discussion with @gctaylor

* bump version

Commit:
Add sleep after ns so resources can be cleaned properly (#2992)

benjigoldberg committed:
Kafka chart use confluent kafka (#2988)

* Use Confluent OSS Kafka Image, Allow user configuration overrides, use a true liveness check

* Add Test for Kafka Chart that creates a topic, produces and consumes todays date

* Remove superfluous producer line from test

Commit:
odoo: update to `bitnami/odoo:10.0.20171212-r0`

Commit:
odoo: bump chart appVersion to `10.0.20171212`

Commit:
odoo: bump chart version to `0.6.5`

pulledtim committed:
Merge branch 'master' into gsqlNameOverride

timothyclarke committed:
Merge branch 'master' into master

pulledtim committed:
Merge branch 'master' into gsqlNameOverride

Commit:
Merge branch 'master' into patch-2

Commit:
joomla: update to `bitnami/joomla:3.8.3-r0`

Commit:
joomla: bump chart appVersion to `3.8.3`

Commit:
joomla: bump chart version to `0.5.4`

Commit:
Charlyf/datadog/updateto5172 (#2474)

* adding hostport to ds and svc

* adding leader election to the Datadog Chart

* edits per @hkaj's review

* Tuning RBAC to the standard config and to the LeaderElection setup

* rebasing on master, correcting README inconsistency and adding comments

* adding leader election to the Datadog Chart

* edits per @hkaj's review

* rebasing on master, correcting README inconsistency and adding comments

* adding leader election to the Datadog Chart

* edits per @hkaj's review

* Tuning RBAC to the standard config and to the LeaderElection setup

* adding leader election to the Datadog Chart

* edits per @hkaj's review

* rebasing on master, correcting README inconsistency and adding comments

* rebasing on master, correcting README inconsistency and adding comments

* removing comment causing e2e check to fail

yuanying committed:
[stable/owncloud] Support tls termination and existingClaim (#2879)

* [stable/owncloud] Support existingClaim

* [stable/owncloud] Support TLS termination

mattfarina committed:
Fixing OWNERS file to have correct naming conventions (#3014)

mattfarina committed:
Merge branch 'master' into joomla-0.5.4

mattfarina committed:
Merge branch 'master' into odoo-0.6.5

cblecker committed:
Add debugging output to find PV errors in CI

Commit:
Fix viglesiasce Github ID in OWNERS file

Commit:
Merge branch 'master' into fix-viglesiasce-own

k8s-ci-robot committed:
Merge pull request #2997 from bitnami-bot/odoo-0.6.5

[stable/odoo] Release 0.6.5

mattfarina committed:
Merge branch 'master' into joomla-0.5.4

k8s-ci-robot committed:
Merge pull request #3008 from bitnami-bot/joomla-0.5.4

[stable/joomla] Release 0.5.4

Commit:
Merge branch 'master' into fix-viglesiasce-own

k8s-ci-robot committed:
Merge pull request #3018 from viglesiasce/fix-viglesiasce-own

Fix viglesiasce Github ID in OWNERS file

tompizmor committed:
[stable/ghost] Change default ghost port (#2460)

* Fix ghostPort configuration

Bump chart version

* Bump chart version

* Fix undefined variable

* Fix ghost-cli issue and port config

* Fix typo

* add `serviceType` argument to `helm upgrade` command

Commit:
[stable/magento] Release 0.5.2 (#3019)

* magento: update to `bitnami/magento:2.2.2-r0`

* magento: bump chart appVersion to `2.2.2`

* magento: bump chart version to `0.5.2`

pulledtim committed:
Merge branch 'master' into gsqlNameOverride

timothyclarke committed:
Merge branch 'master' into master

timothyclarke committed:
stable/wordpress bumped mariadb version

k8s-ci-robot committed:
Merge pull request #2605 from timothyclarke/master

[stable/wordpress] Optional MariaDB for WordPress

pulledtim committed:
Merge branch 'master' into gsqlNameOverride

mattfarina committed:
Adding OWNERS file for openvpn chart

mattfarina committed:
Adding OWNERS file for drupal chart

mattfarina committed:
Adding review process docs

Commit:
Merge branch 'master' into patch-1

k8s-ci-robot committed:
Merge pull request #3017 from cblecker/patch-1

Add debugging output to find PV errors in CI

Commit:
Add mechanism for cc'ing maintainers + nit commas

Commit:
Merge branch 'master' into guidelines/process

k8s-ci-robot committed:
Merge pull request #3029 from mattfarina/guidelines/process

Adding review process docs

cblecker committed:
Fix junit artifacts output dir

k8s-ci-robot committed:
Merge pull request #3030 from cblecker/artifacts-fix

Fix junit artifacts output dir

cblecker committed:
Use proper kubectl path

k8s-ci-robot committed:
Merge pull request #3032 from cblecker/fix-kubectl

Use proper kubectl path

cblecker committed:
Fix kubectl/helm paths for real

k8s-ci-robot committed:
Merge pull request #3034 from cblecker/fix-kubectl-2

Fix kubectl/helm paths for real

Commit:
[stable/ghost] Release 2.1.4 (#3023)

* ghost: update to `bitnami/ghost:1.19.0-r0`

* ghost: bump chart appVersion to `1.19.0`

* ghost: bump chart version to `2.1.4`

mattfarina committed:
Adding OWNERS file for chartmuseum

k8s-ci-robot committed:
Merge pull request #3028 from mattfarina/owners/drupal-2

Adding OWNERS file for drupal chart

mattfarina committed:
Adding an owners file for datadog

k8s-ci-robot committed:
Merge pull request #3026 from mattfarina/owners/openvpn

Adding OWNERS file for openvpn chart

k8s-ci-robot committed:
Merge pull request #3037 from mattfarina/owners/chartmuseum

Adding OWNERS file for chartmuseum

k8s-ci-robot committed:
Merge pull request #3043 from mattfarina/owners/datadog

Adding an owners file for datadog

Commit:
Update Chart.yaml

Commit:
Merge branch 'master' into master

Commit:
Merge branch 'master' into master

k8s-ci-robot committed:
Merge pull request #2649 from jslusher/master

[elasticsearch] add podAnnotations value

k8s-ci-robot committed:
Merge pull request #2950 from chipironcin/patch-2

Update README

k8s-ci-robot committed:
Merge pull request #2773 from Tim-Smyth/gsqlNameOverride

Make value nameOverride overide name completely

Mashine committed:
Update Chart.yaml

yuvipanda committed:
[stable/grafana] Allow setting extra env variables on server (#2903)

* [stable/grafana] Allow setting extra env variables on server

This allows keeping secret config items (such as auth keys)
separate from non-secret config items by setting them in env variables.

* [stable/grafana] Bump chart version

* [stable/grafana] Try to fix tests

rbsilva committed:
[stable/chaoskube] add RBAC support (#2834)

* [stable/chaoskube] add RBAC support

* [stable/chaoskube] add RBAC support - fix newline

* [stable/chaoskube] add RBAC support - bump version

* [stable/chaoskube] add RBAC support - bump version

* [stable/chaoskube] add RBAC support - fix versions

Commit:
Add a new stable chart "weave-scope". (#1719)

shohhei1126 committed:
[stable/nginx-ingress] fix render and validation errors in nginx-ingress template (#3033)

* fix render and validation errors in nginx-ingress template

* bump up nginx-ingress version

Mashine committed:
include fullname in sentinel metadata name

k8s-ci-robot committed:
Merge pull request #2873 from mashine/patch-1

Prefix with release name to prevent name collision

benmathews committed:
template NOTES and fix hpa for nginx-ingress (#3046)

* return ingressClass in NOTES

* fix broken controller-hpa

* Bump chart version

miekg committed:
CoreDNS chart: update to latest version (#2771)

jar361 committed:
[incubator/logstash] Add logstash Chart (#2952)

tmak committed:
[stable/traefik] Allow enforcing SSL without having SSL enabled in Traefik itself (#2778)

* Allow to enforce SSL without having SSL enabled

* Bump Traefik Chart version to 1.4.3

* Always use regex redirect for enforcing SSL

Commit:
[stable/postgresql] add annotations to pvc (#2858)

kfox1111 committed:
Allow adding pod labels to the daemonset as well. (#2933)

* Allow adding pod labels to the daemonset as well.

* Bump version.

* Bump version.

smileisak committed:
[stable/redis-ha] Update redis image (version 4.0.6) (#3001)

fatsheep9146 committed:
Fix bugs for prometheus chart (#2915)

Signed-off-by: Ziqi Zhao <zhaoziqi@qiniu.com>

Commit:
[external-dns] Fix naming in condition for secret (#2998)

ledor473 committed:
Adding a way to disable the Agent DaemonSet (to let the user run it as a sidecar container) (#2856)

dstrebel committed:
updated chart for Istio 0.2.12 (#2910)

alexanderdavidsen committed:
[stable/nginx-ingress] allow override of StatusExporter.service.targetPort (#2865)

Commit:
Charlyf/dup hostport (#3022)

* adding hostport to ds and svc

* removing hostPort in svc and ds

Commit:
Fix variable path in Grafana set-datasource job. (#3003)

* Fix variable path in Grafana set-datasource job.

* Bump Grafana chart version.

rawlingsj committed:
add optional annotations to chartmuseum kubernetes service yaml (#3024)

* add optional annotations to kubernetes service

* bump chart version

matthope committed:
[stable/concourse] Add ability to set annotations on web service. (#2982)

etiennetremel committed:
[stable/verdaccio] Upgrade version, add logo and fix documentation (#2943)

* Upgrade Verdaccio version, add logo and fix documentation

* Bump to 2.7.0

* Swith logo from SVG to PNG

sfriedel committed:
[stable/grafana] Allow customization of readinessProbe via values.yaml (#2923)

* [stable/grafana] Allow customization of readinessProbe via values.yaml

* [stable/grafana] Add `server.readinessProbe` documentation to README

* [stable/grafana] Bump chart version

anandkumarpatel committed:
add optional image pull secrets (#3015)

* add optional image pull secrets

* Update Chart.yaml

ramakuka committed:
Support to add-header configmap for Nginx-ingress (#2902)

thuandt committed:
[stable/prometheus] Seperate between prefix and external url (#3036)

* Seperate between prefix and external url

+ `baseURL`: external url for access services
+ `prefixURL`: prefix url for kubernetes internal access

* stable/prometheus: increase chart version

diemol committed:
Upgrade to 3.8.1 (#3045)

mattfarina committed:
Adding a version comparison to the k8s CI tests

The bot that merges PRs re-runs the e2e job before merging a pull
request. By also having it here (circleci provides fast response
to PR authors) we make sure the version has been incremented in
the chart prior to it being merged.

Commit:
Double retry count for Jenkins test

Commit:
Merge branch 'master' into jenkins-test-timeout

Commit:
Update Chart.yaml

k8s-ci-robot committed:
Merge pull request #3057 from viglesiasce/jenkins-test-timeout

[stable/jenkins] Double retry count for Jenkins test

k8s-ci-robot committed:
Merge pull request #3056 from mattfarina/ci/version-test

Adding a version comparison to the k8s CI tests

matthope committed:
[stable/concourse] Add "worker.podManagementPolicy" (#2983)

* [stable/concourse] Add "worker.podManagementPolicy"

* Switched workers.podManagementPolicy to default to "Parallel" which is more useful for Concourse workers.

* Removed `default "Parallel"` as this is included in the Values file, as per https://github.com/kubernetes/charts/pull/2983#discussion_r157244521

* Removed conditional that required k8s 1.7+ -- https://github.com/kubernetes/charts/pull/2983#discussion_r157244764

* removed extraneous blank line

axdotl committed:
[stable/influxdb] passing environment variables to influx (#3007)

Commit:
Sentry chart not using mail.use_tls and mail.from_address (#2569) (#3038)

* Sentry chart not using mail.use_tls and mail.from_address (#2569)

* Chart version bump (#2569)

* updated settings in templates/hooks

* Don't define default values for new settings

diemol committed:
Updating maintainers, sorry @flah00 didn't mean to be rude. (#3059)

mattfarina committed:
Fixing edge case for patch version of 0 (#3063)

When the array splitting happen there are cases where the patch
(maj.min.patch) is 0. Using ++ to increment that 0 causes a
return code of 1. Using +=1 instead increments the value but has
a return code of 0.

unguiculus committed:
Double test timeout (#3048)

unguiculus committed:
[incubator/keycloak] Add high availability and various improvements (#2756)

* [incubator/keycloak] Add high availability and various improvements

* Add various improvements, such as node selector, tolerations, and pod anti-affinity
* Switch to StatefulSet due to length restrictions for WildFly node identifier and
  for potential advantages for cluster discovery when names pod stable
* Add high availability and clustering using JDBC_PING discovery for JGroups
* Add customizability via WildFly CLI
* Improve ingress configuration

* Fix NOTES.txt

* Improve external database configuration

* Fix typos

* Add login test

* Fix Postgres conditionals

* Automatically enable HA when keycloak.replicas > 1

octete committed:
incubator/oauth-proxy Allow path to be configured (#2936)

* incubator/oauth-proxy Allow path to be configured

Allow path to be configured on the ingress. If one has a vhost that
needs to be shared with oauth proxy, then we need to be able to proxy
pass just /oauth2 to the proxy. Thus allowing it to be configured here.

* Update Chart version

* Fix variable inside range loop.

ianburrell committed:
Use Values.uuid for cluster id with default of uuidv4 (#3055)

* Use Values.uuid for cluster id with default of uuidv4

Currently uses uuidv4 for cluster-id and the "default" Values.uuid won't be
used.

* Bump spartakus version to 1.1.2

ianmaddox committed:
Redis-ha fixes (#2977)

* Redis-ha fixes

Streamlining init process to only require one kind of Redis server in addition to the Sentinels. The Redis instances deterministically pick their own master now.
Depends on quay.io/ianmaddox/redis:4.0.6
Tightening outage detection
Adding labels for diagnostics
Using Redis 4.0.6
Making Sentinels responsible for moving Redis boxes in/out of master LB on promotion/demotion using labels
Chart version now 1.1.0

* Update README.md

* Adding nodeSelector and dynamic service name handling

* Making services select on release

* Updating image to 4.0.6r0

* PR fixups
Added flag to enable VPC mode on deploy
Moved annotations to values files

* Changing service type declaration to explicit terms instead of mode shifting

* Improved appliance mode docs, annotations, file names

* Cleaning up

* Fix typo

* Removing redundancy in PVC name

* Removing appliance verbiage. Switching image source back. Improving docs

* Removing PVC per PR

* Updating redis-ha appliance install command in readme

* Update default image in README

* Update Redis image in values

* Format fix on empty annotations value

mattfarina committed:
Adding process to become a chart OWNER/approver to the README.md (#3076)

nexeck committed:
[stable/concourse] Add rbac to concourse worker (#2922)

* Add rbac to concourse worker

* Switch to rbac.authorization.k8s.io/v1

* Enable rbac per default

See: https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

* Set default rbac version to v1beta1

* Update Chart.yml

ReSearchITEng committed:
readme typo fix and add --purge to helm delete (#3039)

* typo fix and add --purge to helm delete

typo fix and add --purge to helm delete

* removing the --purge as requested by maintainer.

* Update Chart.yaml

alejandroEsc committed:
[stable/etcd-operator] changes to account for current v0.7.0  (#2826)

* major changes included to allow for backup, restore, and etcd operator deployments. This also includes deployments allowing for tls. Should be easier to understand what each piece means now.

helm lint ran

almost got restore, complete, for some reason container cannot conntact the right service.

renamed items, fixed restore svc and got it working properly.

* fixed naming issue.

* updated deployments for restore and back up and updates readme and some adjustments to values.yaml

* reduced empty lines in values.yaml file

* updated values so as not to try to deploy etcd cluster resource by default.

* changes based on PR comments and rbac standards.

* added names to backup and restore operators

* readmefile updated

* updated documentation regarding rbac

* changed serviceaccount fields for deployments to use template full name or name found in values file.

* rback service name changes revisited

eldada committed:
Better resource management for Artifactory (#3021)

* Better resource management for Artifactory (#1)

* Update Artifactory version
* Add resources requests and limits to all pods
* Ability to set whitelist on Nginx LoadBalancer service
* Support setting a custom artifactory service name (that will also be set in nginx config)
* Add new parameters to README
* Fix artifactory replicas reference. Hardcode database replicas to 1
* Use resource memory requests and limits in Artifactory java options
* Add readinessProbe to Artifactory
* Add readme on creating a Kubernetes Docker registry secret and passing it to helm

* Update Artifactory Chart version to 6.2.3

* Comment out the default resources and update README on how to use them

* Set Artifactory version 5.6.3

* Remove default values for Nginx resources from README

Commit:
[stable/mongodb] Release 0.4.21 (#3073)

* mongodb: update to `bitnami/mongodb:3.6.0-r1`

* mongodb: bump chart version to `0.4.21`

bryanlarsen committed:
etcd-operator: namespace is needed in to interact with cluster (#3053)

* etcd-operator: namespace is needed in to interact with cluster

* Update Chart.yaml

mattfarina committed:
Updated the stale issue/pr handling in the README (#3054)

This reflects the k8s org wide handling recently instituted

JnyJny committed:
Update README.md (#3035)

* Update README.md

The descriptions for `config.cluster_name`, `config.cluster_size` and `config.seed_size` have been updated to more correctly identify what these parameters are used for.

* Update Chart.yaml

smelchior committed:
fix typo for template name (#3070)

mdub committed:
Fix NOTES.txt for kafka chart (#3064)

* Fix intructions for kafka chart.

It appears demo scripts are in "/usr/bin", and have no ".sh" extension.

* Update Chart.yaml

hairmare committed:
[stable/sonatype-nexus]: Do not set targetPort for ClusterIP  (#3069)

* [stable/sonatype-nexus] No targetPort for ClusterIP services

* [stable/sonatype-nexus] Bump chart version

2color committed:
correct typo in comment (#3077)

* correct typo in comment

* Update Chart.yaml

* Update Chart.yaml

komljen committed:
Fix wrong var name in NOTES when Ingress is enabled (#3060)

* Fix wrong var name in NOTES when Ingress is enabled

* Increase Chart version

* Check if tls is enabled and set the URL accordingly

mattfarina committed:
Handling the case of a new chart in the e2e tests (#3083)

etiennetremel committed:
[stable/rabbitmq-ha] Add RabbitMQ HA chart (#3005)

* Add RabbitMQ HA chart

* Use RabbitMQ Alpine image

* Add RBAC support

* Add EPMD port for cross cluster communication, bump default memory watermark to 256MB

* Move serviceAccountName into rbac

* Use Github username

* Fix app label to match rabbitmq-ha.name

* Remove trailing whitespace

Commit:
[incubator/kubewatch] add kubewatch chart (#2705)

* [incubator/kubewatch] add kubewatch chart

* Add missing quote

* iterate chart number

Commit:
[stable/phabricator] Release 0.5.4 (#3088)

* phabricator: update to `bitnami/phabricator:2017.50.0-r0`

* phabricator: bump chart appVersion to `2017.50.0`

* phabricator: bump chart version to `0.5.4`

rawlingsj committed:
avoid helm lint errors when adding annotations in a composable chart … (#3025)

* avoid helm lint errors when adding annotations in a composable chart values.yaml

without this we get a helm lint warning of:
`warning: destination for annotations is a table. Ignoring non-table value <nil>`

* Incrementing chart version

sathieu committed:
[stable/rocketchat] Allow to specify ingress secret (including a dash to unset it) (#2970)

sathieu committed:
[stable/rocketchat] Give more time for Rocket.Chat to start (#2971)

cw-sakamoto committed:
[stable/datadog] Add dependency chart overriding values (#2969)

* add dependency chart overriding values for kube-state-metrics

* delete empty-lines

* modified README

* modified README

* Increment chart version

matthope committed:
[stable/concourse] Fix typo in minimum k8s version for StatefulSet upgradeStrategy (#3117)

joshsouza committed:
[stable/datadog] Tag pods for deployments service to only use deployment (#2702)

mattfarina committed:
Adding OWNERS file for wordpress chart (#3092)

mattfarina committed:
Adding OWNERS file for testlink (#3093)

mattfarina committed:
Adding OWNERS file to redis chart (#3094)

mattfarina committed:
Adding OWNERS file to prestashop chart (#3095)

spiffxp committed:
Add code-of-conduct.md (#3122)

Refer to kubernetes/community as authoritative source for code of conduct

Commit:
[stable/mariadb] Release 2.1.2 (#3116)

* mariadb: update to `bitnami/mariadb:10.1.29-r2`

* mariadb: bump chart version to `2.1.2`

Commit:
[stable/odoo] Release 0.6.6 (#3134)

* odoo: update to `bitnami/odoo:10.0.20171222-r0`

* odoo: bump chart appVersion to `10.0.20171222`

* odoo: bump chart version to `0.6.6`

unguiculus commented:
ping @timstoop

gyliu513 committed:
[stable/dokuwiki] Use storageClassName for dokuwiki. (#1988)

Commit:
[stable/suitecrm] Release 0.3.2 (#3119)

* suitecrm: update to `bitnami/suitecrm:7.9.8-r0`

* suitecrm: bump chart appVersion to `7.9.8`

* suitecrm: bump chart version to `0.3.2`

gyliu513 committed:
[stable/ghost] Use storageClassName for ghost. (#1991)

gyliu513 committed:
[stable/jasperreports] Use storageClassName for jasperreports. (#1996)

gyliu513 committed:
[stable/mediawiki] Use storageClassName for mediawiki. (#2009)

Commit:
[stable/owncloud] Release 0.5.3 (#3126)

* owncloud: update to `bitnami/owncloud:10.0.4-r0`

* owncloud: bump chart appVersion to `10.0.4`

* owncloud: bump chart version to `0.5.3`

Commit:
[stable/redis] Release 1.1.4 (#3139)

* redis: update to `bitnami/redis:4.0.6-r1`

* redis: bump chart version to `1.1.4`

a-robinson committed:
Bump CockroachDB chart's version and other small improvements to it (#3105)

* Add options for configuring cache and SQL memory to CockroachDB chart

* Bump CockroachDB version to v1.1.3

* Improve CockroachDB's PodDisruptionBudget

Letting only 1 go down is much safer and more accurately reflects the
desire of operators than allowing up to 33% to go down. It's
unfortunate that this bumps the minimum version by so much, but 1.7 is a
full two releases ago.

christian-roggia committed:
[incubator/logstash] Add README file and change service type (#3061)

* Added missing nodeSelector in values.yaml

* Changed service.type to ClusterIP by default

* Added README file

* Updated chart information and version

kongslund committed:
stable/mysql: Make liveness and readiness probes configurable (#3068)

* stable/mysql: Make liveness and readiness probes configurable

* Update Chart.yaml

mazen-ammar committed:
[stable/sumologic-fluentd] Update env variables (#3107)

* Added new env variables to daemonset config

* Quote boolean env variables to allow setting via command line

* Add default values for new env variables

* Updated default image version to 1.6

* Updated README with descriptions for new env variables

* Removed trailing whitespace

* Incremented chart version to 0.2.1

* Update README.md

etiennetremel committed:
[stable/rabbitmq-ha] Use ClusterIP as default service type (#3102)

* Use ClusterIP as default service type

* Add documentation for upgrading RabbitMQ chart

* Bump chart version

* Add missing base64 --decode flag

etiennetremel committed:
Upgrade Verdaccio to 2.7.1 (#3115)

jonaseck2 committed:
Switched to environment variable arguments (#3118)

* Switched to environment variable arguments

Environment variables are evaluated before command-line arguments causing parsing of SPARK_MASTER_PORT to crash if left to the default value set by kubernetes.

* Fixed lint errors: missing starting space in comment

marckhouzam committed:
[stable/grafana] Make docker image tag fixed (#2480) (#3132)

The current Grafana chart is not immutable because it uses the
grafana/grafana:latest image.

In my particular case, I need to be able to deploy Grafana one 
day and know that when I deploy it the next day, I will get the
same version. That way I can deploy to many environments over a
period of multiple days.

This PR locks the image version to the current latest
grafana/grafana:4.6.3

Of course, with this change, it becomes essential to make new
versions of the chart when a new version of Grafana becomes
available. This is the approach taken by many charts.

lnguyen committed:
[incubator/zookeeper] adding additional affinities (#2907)

* [incubator/zookeeper] adding additional affinities

* Removed antiaffinities and support just affinities in general

* Fixed to default affinity as empty hash

* fixing extra end

timstoop commented:
Apologies, I've been very busy :( I'll pick this up again next week, after Christmas.

tompizmor committed:
[stable/phpbb] allow use of external database (#2815)

* Allow phpBB helm chart to use an existing database

* Add missing env var

* Refactor properties to configure external database

* Fix lint

* Update dependencies

* Avoid exposing the database password for external database setup

* Apply Sameer feedback

* More feedback

* Fix typo

* Fix lint error

tompizmor committed:
[stable/mediawiki] allow use of external database (#2814)

* Allow MediaWiki helm chart to use an existing database

* Add missing env var

* Fix typo

* Refactor properties to configure external database

* Update dependencies

* Avoid exposing the database password for external database setup

* Add Sameer feedback

* More feedback

* Fix lint error

Commit:
[stable/wordpress] Release 0.7.9 (#3152)

* wordpress: update to `bitnami/wordpress:4.9.1-r1`

* wordpress: bump chart version to `0.7.9`

Commit:
[stable/mongodb] Release 0.4.22 (#3153)

* mongodb: update to `bitnami/mongodb:3.6.1-r0`

* mongodb: bump chart appVersion to `3.6.1`

* mongodb: bump chart version to `0.4.22`

Commit:
[stable/mariadb] Release 2.1.3 (#3155)

* mariadb: update to `bitnami/mariadb:10.1.30-r1`

* mariadb: bump chart appVersion to `10.1.30`

* mariadb: bump chart version to `2.1.3`

Commit:
[stable/rabbitmq] Release 0.6.14 (#3004)

* rabbitmq: update to `bitnami/rabbitmq:3.7.0-r0`

* rabbitmq: bump chart appVersion to `3.7.0`

* rabbitmq: bump chart version to `0.6.14`

* rabbitmq: update to `bitnami/rabbitmq:3.7.1-r0`

* rabbitmq: bump chart appVersion to `3.7.1`

* rabbitmq: update to `bitnami/rabbitmq:3.7.2-r0`

* rabbitmq: bump chart appVersion to `3.7.2`

* rabbitmq: env var `RABBITMQ_ERLANG_COOKIE` renamed to `RABBITMQ_ERL_COOKIE`

* rabbitmq: `rabbitmqDiskFreeLimit` should be lower than `rabbitmqDiskFreeLimit`

mattfarina committed:
Add more verbose output when helm test fails (#2750)

This change does 3 things:
1. Gets the logs from the test pods
2. Prints the output of helm status
3. Gets the logs from the pods running for a release

mattfarina committed:
Fix broken end to end tests resulting from cleanup code issues (#3168)

mattfarina committed:
Convert registry to k8s.gcr.io (#3160)

stanislavb committed:
[stable/artifactory] Service annotations (#3090)

Allow full customization of service annotations for all three services
in artifactory chart: artifactory, postgres and nginx. Purpose of the
change is to allow end user to provide configuration for other
kubernetes components to act on.

mattfarina committed:
Convert registry to k8s.gcr.io (#3158)

See #3141 for more detail

mattfarina committed:
Convert registry to k8s.gcr.io (#3159)

mattfarina committed:
[stable/kube-state-metrics] Convert registry to k8s.gcr.io (#3163)

* Convert registry to k8s.gcr.io

* fix hardcoded version in readme

mattfarina committed:
Convert registry to k8s.gcr.io (#3162)

mattfarina committed:
Convert registry to k8s.gcr.io (#3164)

mattfarina committed:
Convert registry to k8s.gcr.io (#3166)

mattfarina committed:
Convert registry to k8s.gcr.io (#3169)

mattfarina committed:
Convert registry to k8s.gcr.io (#3170)

mattfarina committed:
Convert registry to k8s.gcr.io (#3171)

mattfarina committed:
Convert registry to k8s.gcr.io (#3161)

valo committed:
Delete jobs using kubectl, not helm (#3176)

* Delete jobs using kubectl, not helm

The jobs should be deleted using `kubectl` not `helm`.

* Updated version

nitisht committed:
Bump minio chart to latest release RELEASE.2017-12-28T01-21-00Z (#3177)

matthope committed:
[stable/concourse] Add support for exposing prometheus metrics. (#3125)

In concourse, to expose prometheus metrics, need to set the env vars:
  CONCOURSE_PROMETHEUS_BIND_IP = 0.0.0.0
  CONCOURSE_PROMETHEUS_BIND_PORT = portnumber

This provides an HTTP endpoint to scrape the metrics.

Enabling metrics via the values.yaml is half the battle, still need to configure prometheus (potentially via service discovery and annotations) to scrape the data.

goruha committed:
Added new version of chartmuseum (#3050)

* Added new version of chartmuseum

* Fix linting

* Address PR  comments

* Added multiple ingress based on values.yaml

* Added Owners

* Fix circle CI

* Bump version

* Fix deployment spec

* Make ingress single resource

* Fix Owners

* Fix values.yaml

* Bump version

gianarb committed:
Bump InfluxDB (#3186)

/assign @mattfarina

gianarb committed:
Bump Chronograf (#3187)

/assign @mattfarina

gianarb committed:
Bump Telegraf (#3188)

/assign @mattfarina

gianarb committed:
Bump Kapacitor (#3189)

/assign @mattfarina

This PR switch Kapacitor to alpine to maintain the same base image
(alpine) across all the tick stack.

It also switch to docker.io for the same reason.

Commit:
[stable/odoo] Release 0.6.7 (#3196)

* odoo: update to `bitnami/odoo:10.0.20180102-r0`

* odoo: bump chart appVersion to `10.0.20180102`

* odoo: bump chart version to `0.6.7`

Commit:
[stable/phabricator] Release 0.5.5 (#3198)

* phabricator: update to `bitnami/phabricator:2017.52.0-r0`

* phabricator: bump chart appVersion to `2017.52.0`

* phabricator: bump chart version to `0.5.5`

Commit:
[stable/mediawiki] Release 0.6.1 (#3199)

* mediawiki: update to `bitnami/mediawiki:1.30.0-r0`

* mediawiki: bump chart appVersion to `1.30.0`

* mediawiki: bump chart version to `0.6.1`

christian-roggia committed:
[incubator/drone] Added data persistence (#3185)

* Updated drone to version 0.8.2

* Added persistence to the server

* Added icon to the chart

* Bumped MINOR version instead of PATCH version

rgreinho committed:
[stable/mysql] Add option to specify service type (#3079)

The MySQL chart only allowed the service type to be ClusterIP,
preventing its use with Minikube. This patch adds the ability to create
the service with the `NodePort` type.

Drive-by:
* Update the `NOTES.txt` file to provide similar instructions to the
PostreSQL chart.

Commit:
Merge branch 'add_proxy_support' of github.com:timstoop/charts into add_proxy_support

timstoop commented:
Darn it, this is not at all what I intended to do... Just wanted to rebase my branch! Closing this one and creating a new pull request, with all the comments resolved as much as I can.
